Source/WebCore/rendering/RenderAttachment.cpp

```diff
@@ -97,8 +97,18 @@ void RenderAttachment::layout()
         layoutShadowContent(newIntrinsicSize);
 }
 
-LayoutUnit RenderAttachment::baselinePosition(FontBaseline, bool, LineDirectionMode, LinePositionMode) const
+LayoutUnit RenderAttachment::baselinePosition(FontBaseline fontBaseline, bool firstLine, LineDirectionMode lineDirectionMode, LinePositionMode linePositionMode) const
 {
+    if (auto* baselineElement = attachmentElement().wideLayoutElementForBaseline()) {
+        bool unusedIsReplaced;
+        auto attachmentRect = attachmentElement().pixelSnappedRenderRect(&unusedIsReplaced);
```

Comment:
any particular reason why you consult the DOM tree for geometry (as opposed to the render tree)?

Reply:
The render code doesn't know what these sub-elements mean.